abopt/testing/__init__.py

```diff
@@ -2,6 +2,41 @@
 from scipy.optimize import rosen, rosen_der, rosen_hess_prod, rosen_hess
 import numpy
 
+class ChiSquareProblem_dual(Problem):
+    """
+        chisquare problem with
+
+        y = |J phi(x) - 1.0|^2
+
+        adding on same time evaluation of gradient and forward pass
+    """
+    def __init__(self, J, phi=lambda x:x, phiprime=lambda x: 1, precond=None):
+        def f(x): return J.dot(phi(x))
+        def vjp(x, v): return v.dot(J) * phiprime(x)
+        def jvp(x, v): return J.dot(v * phiprime(x))
+
+        def objective(x):
```

Comment:
Shall we do a test for each of the following cases:

1. only objective_gradient
2. objective and objective_gradient
3. objective and gradient

Reply:
Will do, I have also added a statement that checks that objective and gradient are not None is objective_gradient is.